### DISCUSSION TITLE
Is it possible to add a NOT NULL constraint to a column of an existing table?

### DISCUSSION THREAD

**Discussion by parvizfarnia** [duckdb/duckdb#7687] [Q&A] [answered]
Hello everyone,

I would like to know whether it is possible in DuckDB to add a NOT NULL constraint to a column later after loading a CSV file into the table.

For example, let's say that we have the following table:

`create table employees(employee_id varchar, city_code varchar)`

We insert a CSV file containing the values for the two above-mentioned columns into `employees` table.

At the end we wish to add a `NOT NULL` constraint (after inserting the data) to `city_code` column.

I checked the [ALTER TABLE](https://duckdb.org/docs/sql/statements/alter_table.html) but I didn't find any option for addaing a `NOT NULL` constraint later to a given column in the table.

Can this be done in DuckDB?

Thanks in advance.

**Comment by Mytherin** [comment, marked as answer]
Yes, this can be done:

```sql
CREATE TABLE integers(i INT);
INSERT INTO integers VALUES (NULL);
ALTER TABLE integers ALTER COLUMN i SET NOT NULL;
-- Error: Constraint Error: NOT NULL constraint failed: integers.i
```

I will open an issue to add this to the docs.

**Comment by parvizfarnia** [comment]
Thank you very much for your help!
